pkg/eraser/eraser.go

```
@@ -151,6 +162,45 @@ func removeImages(c Client, targetImages []string) error {
 	return nil
 }
 
+func isExcluded(img string, idToTagListMap map[string][]string) bool {
+	// check if img excluded by digest
+	if _, contains := excluded[img]; contains {
+		return true
+	}
+
+	// check if img excluded by name
+	if len(idToTagListMap[img]) > 0 {
+		if _, contains := excluded[idToTagListMap[img][0]]; contains {
+			return true
+		}
+	}
+
+	r := regexp.MustCompile(`[a-z0-9]+([._-][a-z0-9]+)*\/\*\z`)
+
+	// look for excluded repository values
+	for key := range excluded {
+		// if excluded key ends in /*, check image with pattern match
+		if match := r.MatchString(key); match {
+			// store repository name
+			repo := strings.Split(key, "*")
+
+			// check if img is part of repo
+			if match2, _ := regexp.MatchString("^"+repo[0], img); match2 {
+				return true
+			}
+
+			// retrieve and check by name in the case img is digest
+			if len(idToTagListMap[img]) > 0 {
+				if match2, _ := regexp.MatchString(repo[0], idToTagListMap[img][0]); match2 {
```

See above comments about swallowing errors and regex issues.